### PR TITLE
feat: add set experiment labels and description endpoints [DET-3891 DET-3890]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- add set labels and description path requests.

labels are simplified to a list of strings.

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: proto/src/determined/api/v1/api.proto [hunk 1]
```diff
@@ -167,7 +167,16 @@ service Determined {
...
+    // Set labels for an experiment.
+    rpc SetExperimentLabels(SetExperimentLabelsRequest) returns (SetExperimentLabelsResponse) {
+        option (google.api.http) = {patch: "/api/v1/experiments/{id}/labels" body: "*" };
+        option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {tags: "Experiments"};
+    }
+    // Set description for an experiment.
+    rpc SetExperimentDescription(SetExperimentDescriptionRequest) returns (SetExperimentDescriptionResponse) {
+        option (google.api.http) = {patch: "/api/v1/experiments/{id}/description" body: "*" };
+        option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {tags: "Experiments"};
+    }
```
Comment (stoksc) [inline proto/src/determined/api/v1/api.proto:179]: Coming back to this one again, but do you think this should be a `POST`, `PATCH` or `PUT`. I can see arguments for all - usually I just saw it's an RPC to update a subfield of a resource and that's a `POST`, but I could also see it being a total update of `/description` or `/labels` and a `PUT`.

In other places, I think requests like this have an `POST`s. I'm ok with changing that, but we should go ahead and make a solid decision and stick to it.

**Comment by hamidzr** [comment]
I added a single patch endpoint and there a few points I think would be good to discuss (I added those as comments to the code as well )

File: master/internal/api_experiment.go [hunk 2]
```diff
@@ -16,12 +16,48 @@ import (
...
+		Description: exp.Config.Description,
+		Id:          int32(exp.ID),
+		Labels:      *mapSetToList(&labels),
+		StartTime:   protoutils.ToTimestamp(exp.StartTime),
+		EndTime:     protoutils.ToTimestamp(exp.StartTime),
+		Archived:    exp.Archived,
+		// NumTrials: // requires another db query
+		// Progress: exp.prog, // actor call
+		// Username: , // db query
+		// State: exp.State, // a new mapping
```
Comment (hamidzr) [inline master/internal/api_experiment.go:57]: @stoksc write a separate query to just get Numtrials and username from DB?

File: master/go.sum [hunk 3]
```diff
@@ -193,6 +193,7 @@ github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/denis-tingajkin/go-header v0.3.1 h1:ymEpSiFjeItCy1FOP+x0M2KdCELdEAHUsNa8F+hHc6w=
 github.com/denis-tingajkin/go-header v0.3.1/go.mod h1:sq/2IxMhaZX+RRcgHfCRx/m0M5na0fBt4/CRe7Lrji0=
+github.com/determined-ai/determined v0.12.4 h1:lLASmkwInB+Q58ZmBfx2aDYt71cP6Liya6BYljxROQI=
```
Comment (hamidzr) [inline master/go.sum:196]: remove

**Comment by hamidzr** [comment]
I updated the PR based on what we talked about.

File: master/internal/api_experiment.go [hunk 4]
```diff
@@ -282,6 +282,52 @@ func (a *apiServer) UnarchiveExperiment(
...
+		}
+	}
+
+	type experimentPatch struct {
+		Labels      []string `json:"labels"`
+		Description string   `json:"description"`
+	}
+	patches := experimentPatch{Description: exp.Description, Labels: exp.Labels}
+	// QUESTION if I maks this json or manually generate the json string
+	// we can submit only the necessary parts of the patch and avoid getting the experiment at start.
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply master/internal/api_experiment.go:316]: the update at the end could return it but I didn't see a clean way of getting only the needed parts of the patch.

File: master/static/srv/patch_experiment.sql [hunk 5]
```diff
@@ -0,0 +1,4 @@
+UPDATE experiments e
+SET config = config || $2
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply master/static/srv/patch_experiment.sql:2]: yeah it ended up being simpler than I expected

**Pull request merged**